From bd5b511e4bcfb535ced7fc553193ec3c69b1ec1b Mon Sep 17 00:00:00 2001
From: Les Lee <lesl@google.com>
Date: Mon, 22 Aug 2022 15:18:01 +0800
Subject: [PATCH] [DO NOT MERGE] wifi: Reset to default SAP configuration when
 doing factory reset

Also fixed incorrect the isUserConfiguration value when doing reset to default
SAP configuration.

Bug: 241927115
Test: atest -c FrameworksWifiTests
Change-Id: Ia85808e2928bc347d4ba6823b026b49ed2cdf780
Merged-In: Ia212666450fd9fe3a95393defa2fb56383bab83a
(cherry picked from commit 197faf1486e383ba86681dea6daa90b284e30acd)
Merged-In: Ia85808e2928bc347d4ba6823b026b49ed2cdf780
---
 .../com/android/server/wifi/WifiApConfigStore.java    | 11 ++++-------
 .../java/com/android/server/wifi/WifiServiceImpl.java |  3 +++
 .../android/server/wifi/WifiApConfigStoreTest.java    |  2 +-
 .../com/android/server/wifi/WifiServiceImplTest.java  |  1 +
 4 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/service/java/com/android/server/wifi/WifiApConfigStore.java b/service/java/com/android/server/wifi/WifiApConfigStore.java
index 4e756d4e9..17ebe5ff7 100644
--- a/service/java/com/android/server/wifi/WifiApConfigStore.java
+++ b/service/java/com/android/server/wifi/WifiApConfigStore.java
@@ -161,13 +161,10 @@ public class WifiApConfigStore {
      * and the main Wifi thread (CMD_START_AP).
      */
     public synchronized void setApConfiguration(SoftApConfiguration config) {
-        if (config == null) {
-            config = getDefaultApConfiguration();
-        } else {
-            config = sanitizePersistentApConfig(config);
-        }
-        persistConfigAndTriggerBackupManagerProxy(
-                new SoftApConfiguration.Builder(config).setUserConfiguration(true).build());
+        SoftApConfiguration newConfig = config == null ? getDefaultApConfiguration()
+                : new SoftApConfiguration.Builder(sanitizePersistentApConfig(config))
+                        .setUserConfiguration(true).build();
+        persistConfigAndTriggerBackupManagerProxy(newConfig);
     }
 
     /**
diff --git a/service/java/com/android/server/wifi/WifiServiceImpl.java b/service/java/com/android/server/wifi/WifiServiceImpl.java
index c93c724d5..979caf236 100644
--- a/service/java/com/android/server/wifi/WifiServiceImpl.java
+++ b/service/java/com/android/server/wifi/WifiServiceImpl.java
@@ -4106,6 +4106,9 @@ public class WifiServiceImpl extends BaseWifiService {
             removePasspointConfigurationInternal(null, config.getUniqueId());
         }
         mWifiThreadRunner.post(() -> {
+            EventLog.writeEvent(0x534e4554, "241927115", -1,
+                    "Reset SoftApConfiguration to default configuration");
+            mWifiApConfigStore.setApConfiguration(null);
             mPasspointManager.clearAnqpRequestsAndFlushCache();
             mWifiConfigManager.clearUserTemporarilyDisabledList();
             mWifiConfigManager.removeAllEphemeralOrPasspointConfiguredNetworks();
diff --git a/service/tests/wifitests/src/com/android/server/wifi/WifiApConfigStoreTest.java b/service/tests/wifitests/src/com/android/server/wifi/WifiApConfigStoreTest.java
index fcddf7a4d..eb6abca41 100644
--- a/service/tests/wifitests/src/com/android/server/wifi/WifiApConfigStoreTest.java
+++ b/service/tests/wifitests/src/com/android/server/wifi/WifiApConfigStoreTest.java
@@ -305,7 +305,7 @@ public class WifiApConfigStoreTest extends WifiBaseTest {
         verifyDefaultApConfig(mDataStoreSource.toSerialize(), TEST_DEFAULT_AP_SSID);
         verify(mWifiConfigManager).saveToStore(true);
         verify(mBackupManagerProxy).notifyDataChanged();
-        assertTrue(store.getApConfiguration().isUserConfigurationInternal());
+        assertFalse(store.getApConfiguration().isUserConfigurationInternal());
     }
 
     /**
diff --git a/service/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java b/service/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
index 6727c16da..8d55c694e 100644
--- a/service/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
+++ b/service/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
@@ -5636,6 +5636,7 @@ public class WifiServiceImplTest extends WifiBaseTest {
         // Let the final post inside the |factoryReset| method run to completion.
         mLooper.dispatchAll();
 
+        verify(mWifiApConfigStore).setApConfiguration(null);
         verify(mWifiConfigManager).removeNetwork(
                 openNetwork.networkId, Binder.getCallingUid(), TEST_PACKAGE_NAME);
         verify(mWifiConfigManager).removeNetwork(
-- 
2.39.0.rc1.256.g54fd8350bd-goog

